crates/kitsune_p2p/proxy/benches/thru.rs

```diff
@@ -0,0 +1,211 @@
+                data.extend_from_slice(REQ);
+                con.write(0.into(), data, t).await?;
+                KitsuneResult::Ok(())
+            });
+        }
+
+        futures::future::try_join_all(futs).await.unwrap();
+
+        for _ in 0..self.nodes.len() {
+            d_recv.recv().await;
+        }
+
+        self.d_send
+            .share_mut(|i, _| {
+                *i = None;
+                Ok(())
+            })
+            .unwrap();
+    }
+}
+
+async fn mk_core() -> (TxUrl, Ep, EpHnd) {
+    let t = KitsuneTimeout::from_millis(5000);
+
+    let f = tx2_promote(MemBackendAdapt::new(), NODE_COUNT * 3);
+    let f = tx2_proxy(f, TlsConfig::new_ephemeral().await.unwrap());
+
+    let ep = f.bind("none:", t).await.unwrap();
+    let ep_hnd = ep.handle().clone();
+    let addr = ep_hnd.local_addr().unwrap();
+
+    (addr, ep, ep_hnd)
+}
+
+async fn mk_proxy() -> (TxUrl, EpHnd) {
+    let (addr, mut ep, ep_hnd) = mk_core().await;
+
+    tokio::task::spawn(async move { while let Some(_evt) = ep.next().await {} });
+
+    (addr, ep_hnd)
+}
+
+async fn mk_tgt() -> (TxUrl, EpHnd) {
+    let (addr, mut ep, ep_hnd) = mk_core().await;
+
+    tokio::task::spawn(async move {
+        while let Some(evt) = ep.next().await {
+            if let EpEvent::IncomingData(EpIncomingData { con, mut data, .. }) = evt {
+                let t = KitsuneTimeout::from_millis(5000);
+                if data.as_ref() == REQ {
+                    data.clear();
+                    data.extend_from_slice(RES);
+                    con.write(0.into(), data, t).await.unwrap();
+                } else {
+                    panic!("unexpected bytes");
+                }
+            }
+        }
+    });
+
+    (addr, ep_hnd)
+}
+
+async fn mk_node(d_send: Arc<Share<Option<tokio::sync::mpsc::Sender<()>>>>) -> (TxUrl, EpHnd) {
+    let (addr, mut ep, ep_hnd) = mk_core().await;
+
+    tokio::task::spawn(async move {
+        while let Some(evt) = ep.next().await {
+            if let EpEvent::IncomingData(EpIncomingData { data, .. }) = evt {
+                if data.as_ref() == RES {
+                    let d_send = d_send
+                        .share_mut(|i, _| Ok(i.as_ref().unwrap().clone()))
+                        .unwrap();
+                    d_send.send(()).await.unwrap();
+                } else {
+                    panic!("unexpected bytes");
+                }
+            }
+        }
+    });
+
+    (addr, ep_hnd)
+}
+
+async fn test(this: &Arc<Share<Option<Test>>>) {
+    let mut t = this.share_mut(|i, _| Ok(i.take().unwrap())).unwrap();
+    t.test().await;
+    this.share_mut(move |i, _| {
+        *i = Some(t);
+        Ok(())
+    })
+    .unwrap();
+}
+
+fn thru(t: &Arc<Share<Option<Test>>>) {
+    let _g = RUNTIME.enter();
+    RUNTIME.block_on(async {
+        test(t).await;
+    });
+}
+
+fn criterion_benchmark(c: &mut Criterion) {
+    let t = RUNTIME.block_on(async { Arc::new(Share::new(Some(Test::new().await))) });
```

Comment:
Not a huge deal but `criterion_benchmark` is only ever called once so you don't really need the Lazy and you could just create the runtime in this function.

Reply:
fixed.